modules/hackage-quirks.nix

```diff
@@ -45,25 +45,19 @@ in [
 
   ({config, lib, pkgs, ...}:
     { _file = "haskell.nix/overlays/hackage-quirks.nix#haskell-language-server"; } //
-    lib.mkIf (config.name == "haskell-language-server") {
-      # TODO remove this when `dependent-sum-0.7.1.0` constraint on `some` has been updated.
-      # See https://github.com/haskell/haskell-language-server/issues/2969
-      # and https://github.com/obsidiansystems/dependent-sum/issues/71
-      cabalProject = lib.mkDefault (''
-        packages: .
-        constraints: dependent-sum >=0.7.1.0
-      ''
-      # TODO remove once these the plugins have been updated in hackage
-      + lib.optionalString (config.version == "1.8.0.0") ''
-        package haskell-language-server
-          flags: -qualifyimportednames${
-            # Stylish haskell is broken for GHC 9.2
-            lib.optionalString (__elem config.compiler-nix-name ["ghc921" "ghc922" "ghc923" "ghc924" "ghc925" "ghc926" "ghc927"]) " -stylishhaskell"
-            # Hlint with this HLS only compiles for GHC 9.0
-            + lib.optionalString (!__elem config.compiler-nix-name ["ghc901" "ghc902"]) " -hlint"
-        }
-        constraints: hls-fourmolu-plugin <1.1.1.0, hls-rename-plugin <1.0.2.0, hls-stan-plugin <1.0.1.0
-      '');
+    lib.mkIf (config.name == "haskell-language-server" && config.version == "github-1.10") {
```

Comment:
Er, I think this is going to cause more problems. Nobody downstream can select a different version, we're doing something completely different based on a magic string...

What about letting `tool` take a `src` argument; if it has that, then call `cabalProject` instead. And then it's up to the user to get the HSL source (which is indeed awkward for us to do because we need the SHA, of course).